File: LeetCodeJava/35.搜索插入位置.java
/*
 * @lc app=leetcode.cn id=35 lang=java
 *
 * [35] 搜索插入位置
 */
class Solution {
    public int searchInsert(int[] nums, int target) {
        if (nums.length == 0) {
            return 0;
        }
        int l = 0;
        int r = nums.length - 1;
        int mid;
        while (r - l > 1) {
            mid = (l + r) / 2;
            if (nums[mid] < target) {
                l = mid;
            } else {
                r = mid;
            }
        }
        return target > nums[l] ? target <= nums[r] ? r : r + 1 : l;
    }
}

File: LeetCodePython/35.搜索插入位置.py
#
# @lc app=leetcode.cn id=35 lang=python3
#
# [35] 搜索插入位置
#

# @lc code=start
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if not nums or nums[0] >= target:
            return 0
        l = 0
        r = len(nums)
        while r - l > 1:
            m = (l + r) // 2
            mid = nums[m]
            if mid == target:
                return m
            elif mid < target:
                l = m
            else:
                r = m
        return r

        
# @lc code=end

